Watchman.Discord/Areas/Protection/Strategies/CapslockDetectorStrategy.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Linq;
+using Devscord.DiscordFramework.Framework.Commands.AntiSpam;
+using Devscord.DiscordFramework.Framework.Commands.AntiSpam.Models;
+using Devscord.DiscordFramework.Framework.Commands.Parsing.Models;
+using Devscord.DiscordFramework.Middlewares.Contexts;
+using Watchman.DomainModel.Settings.ConfigurationItems;
+using Watchman.DomainModel.Settings.Services;
+
+namespace Watchman.Discord.Areas.Protection.Strategies
+{
+    public class CapslockDetectorStrategy : ISpamDetector
+    {
+        private readonly IConfigurationService _configurationService;
+        public IUserSafetyChecker UserSafetyChecker { get; set; }
+
+        public CapslockDetectorStrategy(IUserSafetyChecker userSafetyChecker, IConfigurationService configurationService)
+        {
+            this._configurationService = configurationService;
+            this.UserSafetyChecker = userSafetyChecker;
+        }
+
+        public SpamProbability GetSpamProbability(ServerMessagesCacheService serverMessagesCacheService, DiscordRequest request, Contexts contexts)
+        {
+            if (!this.IsMessageWithMuchCapslock(request.OriginalMessage, contexts.Server.Id))
+            {
+                return SpamProbability.None;
+            }
+            var lastFewMessages = serverMessagesCacheService.GetLastUserMessages(contexts.User.Id, contexts.Server.Id)
+                .TakeLast(6)
+                .ToList();
+
+            var capslockedMessagesCount = lastFewMessages.Count(x => this.IsMessageWithMuchCapslock(x.Content, contexts.Server.Id));
+            capslockedMessagesCount++; // now handled messages is also with capslock
```

Comment:
dlaczego nie dopuszczamy opcji z 0 => none
?

Reply:
0 nigdy nie będzie bo na początku metody jest warunek
jeśli nie jest spełniony to oznacza że co najmniej 1 wiadomość jest z casplockiem

Reply:
dziwnie to wygląda jak jest ++
ktoś kto widzi ten kod wyrwany z kontekstu, bez wiedzy co dzieje sie wcześniej
może pomyśleć że sztucznie każdemu zawyżamy szanse na dostanie muta

Reply:
to jak to inaczej?

Reply:
może coś jest źle zaprojektowane
bo `IsMessageWithMuchCapslock`
powinno działać tak
że jak count zwraca zero to nie ma

Reply:
obstawiam że założenia na wyższym poziomie są złe i trzeba teraz robić takie rzeczy żeby działało

ktoś nowy do tego spojrzy i będzie sie zastanawiać czemu to tu jest

Reply:
to jest dlatego, że aktualnie obsługiwana komenda idzie osobno niż wszystkie poprzednie
mogę dodać do lastFewMessages ostatnią wiadomość, po prostu będzie sprawdzana dwa razy

Reply:
czemu idzie osobno?